### PR TITLE
Refactor affine cell proxy

### PR DESCRIPTION
**Pull request by rrahn**
part of #1908 

Some preparations before we add the pairwise matrix which is a combination of the score and the trace matrix. 
The proxy will then be extended to model both versions: only the score matrix and the combined score and trace matrix.

### CODE REVIEW

File: include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp [hunk 1]
```diff
@@ -13,22 +13,56 @@
...
 
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/simd/concept.hpp>
 #include <seqan3/core/tuple_utility.hpp>
 
 namespace seqan3::detail
 {
 
+/*!\interface seqan3::detail::arithmetic_or_simd <>
+ * \brief The concept for an type that models either seqan3::arithmetic or seqan3::simd::simd_concept.
```
Comment (eseiler) [inline include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp:26]:
```suggestion
 * \brief The concept for a type that models either seqan3::arithmetic or seqan3::simd::simd_concept.
```

File: include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp [hunk 2]
```diff
@@ -13,22 +13,56 @@
...
+ * \ingroup alignment_matrix
+ */
+//!\cond
+template <typename t>
+SEQAN3_CONCEPT arithmetic_or_simd = arithmetic<t> || simd_concept<t>;
+//!\endcond
+
+/*!\interface seqan3::detail::affine_score_cell <>
+ * \extends seqan3::tuple_like
+ * \brief The concept for an type that models an affine cell of the score matrix.
```
Comment (eseiler) [inline include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp:36]:
```suggestion
 * \brief The concept for antype that models an affine cell of the score matrix.
```

File: include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp [hunk 3]
```diff
@@ -50,56 +50,69 @@ class affine_cell_proxy : public tuple_t
...
-    //!\brief Access the optimal score of the wrapped score matrix cell.
-    decltype(auto) best_score() & { using std::get; return get<0>(*this); }
+    //!\brief Access the best score of the wrapped score matrix cell.
+    decltype(auto) best_score() & noexcept { return get_score_impl<0>(*this); }
     //!\overload
-    decltype(auto) best_score() const & { using std::get; return get<0>(*this); }
+    decltype(auto) best_score() const & noexcept { return get_score_impl<0>(*this); }
     //!\overload
-    decltype(auto) best_score() && { using std::get; return get<0>(std::move(*this)); }
+    decltype(auto) best_score() && noexcept  { return get_score_impl<0>(std::move(*this)); }
```
Comment (smehringer) [inline include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp:96]:
```suggestion
    decltype(auto) best_score() && noexcept { return get_score_impl<0>(std::move(*this)); }
```

File: include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp (hunk 1 above)
Comment (smehringer) [reply include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp]: not resolved?

File: include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp [hunk 4]
```diff
@@ -13,12 +13,43 @@
...
+//!\endcond
+
+/*!\interface seqan3::detail::affine_score_cell <>
+ * \extends seqan3::tuple_like
+ * \brief The concept for an type that models an affine cell of the score matrix.
+ * \ingroup alignment_matrix
+ *
+ * \details
+ *
+ * This concept describes the requirements an alignment matrix cell must fulfil to represent an affine score
```
Comment (smehringer) [inline include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp:41]:
```suggestion
 * This concept describes the requirements an alignment matrix cell must fulfill to represent an affine score
```

File: include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp [hunk 5]
```diff
@@ -33,7 +33,7 @@ SEQAN3_CONCEPT arithmetic_or_simd = arithmetic<t> || simd_concept<t>;
 
 /*!\interface seqan3::detail::affine_score_cell <>
  * \extends seqan3::tuple_like
- * \brief The concept for an type that models an affine cell of the score matrix.
+ * \brief The concept for antype that models an affine cell of the score matrix.
```
Comment (smehringer) [inline include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp:36]:
```suggestion
 * \brief The concept for a type that models an affine cell of the score matrix.
```

File: include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp (hunk 4 above)
Comment (eseiler) [reply include/seqan3/alignment/matrix/detail/affine_cell_proxy.hpp:41]: Fulfill is AE
Fulfil is  BE

**Pull request merged**